src/intent_parser/lookup_intent_parser.rs

```diff
@@ -0,0 +1,824 @@
+            .custom_entity_parser
+            .extract_entities(input, None)?
+            .into_iter()
+            .map(|entity| entity.into());
+
+        // combine entities
+        let mut all_entities: Vec<MatchedEntity> = Vec::new();
+        all_entities.extend(b);
+        all_entities.extend(c);
+
+        Ok(deduplicate_overlapping_entities(all_entities))
+    }
+
+    fn empty_parsing_result(&self) -> InternalParsingResult {
+        InternalParsingResult {
+            intent: IntentClassifierResult {
+                intent_name: None,
+                probability: 1.0,
+            },
+            slots: vec![],
+        }
+    }
+
+    fn parse_map_output(
+        &self,
+        input: &str,
+        output: Option<&(i32, Vec<i32>)>,
+        entities: Vec<MatchedEntity>,
+        intents: Option<&[&str]>,
+    ) -> InternalParsingResult {
+        if let Some((intent_id, slots_ids)) = output {
+            // assert invariant: length of slot ids matches that of entities
+            debug_assert!(slots_ids.len() == entities.len());
+            let intent_name = &self.intents_names[*intent_id as usize];
+            if intents
+                .unwrap_or(&[intent_name.as_ref()])
+                .contains(&intent_name.as_ref())
+            {
+                let intent = IntentClassifierResult {
+                    intent_name: Some(intent_name.to_string()),
+                    probability: 1.0,
+                };
+                // get slots and return result
+                // we assume entities are sorted by their ranges
+                let mut slots = vec![];
+                for (slot_id, entity) in slots_ids.iter().zip(entities.iter()) {
+                    let slot_name = &self.slots_names[*slot_id as usize];
+                    let entity_name = &entity.entity_name;
+                    let char_range = &entity.range;
+                    let value = substring_with_char_range(input.to_string(), &char_range);
+                    slots.push(InternalSlot {
+                        value,
+                        char_range: char_range.clone(),
+                        entity: entity_name.to_string(),
+                        slot_name: slot_name.to_string(),
+                    });
+                }
+                InternalParsingResult { intent, slots }
+            } else {
+                // if intent name not in intents, return empty result
+                self.empty_parsing_result()
+            }
+        } else {
+            // if lookup value was none return empty result
+            self.empty_parsing_result()
+        }
+    }
+}
+
+impl IntentParser for LookupIntentParser {
+    fn parse(
+        &self,
+        input: &str,
+        intents_whitelist: Option<&[&str]>,
+    ) -> Result<InternalParsingResult> {
+        let entities = self.get_all_entities(input)?;
+        let (_, formatted_input) =
+            replace_entities(input, entities.clone(), get_entity_placeholder);
```

Comment:
A much simpler function than `replace_entities` can be used here, as we don't need to retrieve the ranges.